Engineering/PracticiesAndPrinciples.md

```diff
@@ -18,6 +18,8 @@ While working on the project we try to stick with best practices in architecture
 * Remember to tether to ‘Done’ for a backlog item. Our goal is not to write code, it is to ship incremental customer value. Our job is likely not done until our code is in production adding that value.
 * We all own our code and each one of us has an obligation to make all parts of the system great
 * Keeping backlog item details up to date is how you communicate the state of things with the rest of your team. Discipline around updating status, assignment, comments etc is essential
+* Automate agile workflows using tools like Standup bots (in Microsoft Teams or other communication tools) to "give back" more time to developers for development.
+* The above can also be done towards making retrospectives more efficient by deploying a "suggestion box" bot to capture highlights, lowlights, etc. during the sprint that can be "deposited" during the sprint. This serves the purpose of enhancing the quality of comments because there is no need to think back at the end of the sprint about what happened. It also saves time at the end of the sprint in having to document different points for the retrospective. 
```

Comment:
This might be better stated as "record highlights, lowlights, and major issues during the project to simplify sprint retrospectives". That keeps it generic enough to be used with any customer tools.

We can link to another page that talks about effective retrospectives (Engineering/Retrospectives.md) where a discussion on bots can develop as well as on tools like Team Retrospectives (https://www.microsoft.com/en-us/garage/profiles/team-retrospectives/) which is an Azure Dev Ops extension.

Reply:
I really like the ideas for suggesting mechanisms to facilitate better discussions for status, retros, and planning. But I don't know if I see the value in _prescriptive_ about the specific tools (i.e. bots, plug-ins, extensions) used to manage the agile process. 

Might it be better to have a list of tools teams might want to consider to aid them and specifically, when these tools are useful? For example, when a team is geographically dispersed, the team retro plug-in is more useful. When the team is co-located, good ol' post-it notes are still the king (IMHO).